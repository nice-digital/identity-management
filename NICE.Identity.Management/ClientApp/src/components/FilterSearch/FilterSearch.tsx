import React from "react";
import { Panel } from "@nice-digital/nds-panel";
import { Input } from "@nice-digital/nds-forms";

type FilterSearchProps = {
	onInputChange: (searchQuery: string) => void;
	label: string;
};

export const FilterSearch = (props: FilterSearchProps): React.ReactElement => {
	let typingTimer = 0;

	const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
		const val = e.target.value.trim();

		clearTimeout(typingTimer);

		typingTimer = window.setTimeout(() => {
			props.onInputChange(val);
		}, 1000);
	};

	return (
		<Panel>
			<Input
				type="search"
				label={props.label}
				unique="userSearch"
				name="user-search"
				onChange={handleInputChange}
				autoComplete="off"
			/>
		</Panel>
	);
};
