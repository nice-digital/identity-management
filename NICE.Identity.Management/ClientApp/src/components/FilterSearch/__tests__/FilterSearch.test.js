import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { FilterSearch } from "../FilterSearch";

test("should match the snapshot when querystring search has a value", async () => {
	const dummyText = "SomeText";
	const {container} = render(
		<MemoryRouter initialEntries={[`/organisations?q=${dummyText}`]}>
			<FilterSearch label="Filter by organisation name" />
		</MemoryRouter>,
	);	
	expect(container).toMatchSnapshot();
});