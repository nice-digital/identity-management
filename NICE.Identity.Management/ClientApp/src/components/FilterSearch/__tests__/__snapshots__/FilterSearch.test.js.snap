// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSearch should match the snapshot when querystring search has a value 1`] = `
<div
  className="panel"
>
  <div
    className="input"
  >
    <label
      className="input__label"
      htmlFor="filter-search"
    >
      Filter by organisation name
    </label>
    <input
      autoComplete="off"
      className="input__input"
      data-qa-sel="filter-search-input"
      defaultValue="SomeText"
      id="filter-search"
      name="filter-search"
      onChange={[Function]}
      type="search"
      unique="filterSearch"
    />
  </div>
</div>
`;
